pkg/plugins/golang/v3/scaffolds/api.go

```diff
@@ -83,48 +76,55 @@ func (s *apiScaffolder) newUniverse() *model.Universe {
 
 // TODO: re-use universe created by s.newUniverse() if possible.
 func (s *apiScaffolder) scaffold() error {
-	if s.doResource {
-
-		s.config.UpdateResources(s.resource.GVK())
+	// Check if we need to do the API and controller before updating with previously existing info
+	doAPI := s.resource.API != nil && s.resource.API.Version != ""
```

Comment:
ditto above. We should respect the input. why we start to check s.resource.API.Version != ""/ And the, has no need to `doController := s.resource.Controller` at all?